### PR TITLE
Add test for `const async fn`

### PR DESCRIPTION
**Pull request by Milo123459**
This adds a test for #102796

### CODE REVIEW

File: tests/ui/async-await/const-async/const-async-fn-in-main.rs
Comment (cjgillot) [inline tests/ui/async-await/const-async/const-async-fn-in-main.rs:1]: Does this test need to be in its own directory?

**Review by cjgillot — COMMENTED** [review]
For your future contributions, could you avoid adding an issue number in the PR title? We can't click on it... Adding a summary of the referenced PR's title is much more useful.

File: tests/ui/async-await/const-async/const-async-fn-in-main.rs
Comment (Milo123459) [reply tests/ui/async-await/const-async/const-async-fn-in-main.rs:1]: ive removed the directory

**Comment by compiler-errors** [comment]
please squash this into one commit as well

**Comment by Milo123459** [comment]
done - sorry for not.

**Review by wesleywiser — APPROVED** [review]

**Comment by wesleywiser** [comment]
Thanks @Milo123459!

@bors r+ rollup

**Pull request merged**